frontend/src/stores/calendar-store.ts

```diff
@@ -44,6 +52,48 @@ export const useCalendarStore = defineStore('calendars', () => {
     }
   };
 
+  /**
+   * Get all cremote events from connected calendars for given time span
+   * @param call preconfigured API fetch function
+   * @param activeDate Dayjs object defining the current month
+   * @param force force a refetch
+   */
+  const getRemoteEvents = async (call: Fetch, activeDate: Dayjs, force = false) => {
+    // Get month identifier to remember this month's events are already retrieved
+    const month = activeDate.format('YYYY-MM');
+    
+    // Most calendar impl are non-inclusive of the last day, so just add one day to the end.
+    const from = activeDate.startOf('month').format('YYYY-MM-DD');
+    const to = activeDate.endOf('month').add(1, 'day').format('YYYY-MM-DD');
+  
+    // If retrieval is forced, delete cache and start with zero events again
+    if (force) {
+      remoteMonthsRetrieved.value = [];
+    }
+    const calendarEvents = force ? [] : [...remoteEvents.value];
+
+    // Only retrieve remote events if we don't have this month already cached
+    if (!remoteMonthsRetrieved.value.includes(month)) {
```

Comment:
Can we flip this so that it's like:
```suggestion
    if (remoteMonthsRetrieved.value.includes(month)) {
        return;
    }
    ... existing code but here now ...
```

As I don't see why we would need to re-push an already cached month, but let me know if I've misread this function!

Reply:
Whoops, you're totally right 😅